core/src/main/scala/com/softwaremill/kmq/redelivery/streams/RedeliveryTrackerStream.scala

```diff
@@ -41,10 +42,64 @@ class RedeliveryTrackerStream(markerConsumerSettings: ConsumerSettings[MarkerKey
                 ClosedShape
             })
             .run()
+
+          Future.successful(cancellable)
       }
+      .viaMat(Flow.fromGraph(new AggregatingToMatFlow()))(ControlWithCancellable.apply)
       .toMat(Sink.ignore)(DrainingControl.apply)
       .run()
   }
+}
+
+class AggregatingToMatFlow[T] extends GraphStageWithMaterializedValue[FlowShape[T, T], Iterable[T]] {
```

Comment:
Interesting solution :) @aludwiko or @rucek maybe you can take a look, is this the simplest way to cancel sub-streams created using `.groupBy` or here with `Consumer.committablePartitionedSource`?

Reply:
I hope it's interesting in a positive way :) 

It's certainly not the simplest in general. But here we have additional TickSource merged inside one of the nested sinks which doesn't stop when it's source stops.

Reply:
Yes, in a positive way :)

I think the problem of stopping sub-streams is not that uncommon, as it emerges also when using the `.groupBy` which I mentioned. So either it's already solved in some way, or we have a solution which would be a great material for a short blog post :)